client/lib/signup/step-actions.js

```diff
@@ -358,24 +359,23 @@ function processItemCart(
 	themeSlugWithRepo
 ) {
 	const addToCartAndProceed = () => {
-		let privacyItem = null;
 		const state = reduxStore.getState();
-		const { domainItem } = newCartItems;
 
-		if ( domainItem ) {
-			const { product_slug: productSlug } = domainItem;
-			const productsList = getProductsList( state );
-			if ( supportsPrivacyProtectionPurchase( productSlug, productsList ) ) {
-				privacyItem = updatePrivacyForDomain( domainItem, true );
-
-				if ( privacyItem ) {
-					newCartItems.push( privacyItem );
+		const newCartItemsToAdd = newCartItems.map( item => {
+			// Add privacy protection to domain products, if supported
+			if ( isDomainRegistration( item ) || isDomainTransfer( item ) ) {
+				const { product_slug: productSlug } = item;
+				const productsList = getProductsList( state );
+				if ( supportsPrivacyProtectionPurchase( productSlug, productsList ) ) {
+					return updatePrivacyForDomain( item, true );
```

Comment:
In looking through `productsList`, all products that `supportsPrivacyProtectionPurchase() == true` are `isDomainRegistration() == true`. So it would seem that privacy protection only applies to domain registrations and not to transfers.

However, I also see in the component logic for `DomainDetailsForm` that the privacy section is rendered for domain registrations, or transfers, so I included `isDomainTransfer` here, as well
https://github.com/Automattic/wp-calypso/blob/fix/checkout--domain-details--privacy-protection-default/client/my-sites/checkout/checkout/domain-details-form.jsx#L274

Reply:
Why do you need to check `isDomainRegistration() and `isDomainTransfer()` at all?  It seems to me like you could remove that entire `if()` statement and rely on the `supportsPrivacyProtectionPurchase()` check to exclude non-domain products... That way any future decisions about whether this applies to domain transfers or not can take place inside the data returned by that one method.

Reply:
I had an early version that did exactly this and went back and forth. It seemed problematic to potentially apply a function called `updatePrivacyForDomain` to cart items that are not domain items... and yet it also seems odd to have a somewhat arbitrary gate for before checking if an item `supportsPrivacyProtectionPurchase`

The moment I commited I thought private registration only makes sense for domain registration and transfers, so I'll leave in the check for those, but I'm easily swayed to take it out.

Reply:
>It seemed problematic to potentially apply a function called updatePrivacyForDomain to cart items that are not domain items

The `supportsPrivacyProtectionPurchase` check would prevent that though, right?

It's not a huge deal either way, just seemed to me like the `isDomainRegistration()` and `isDomainTransfer()` are arbitrary gates (as you said), which have the potential to break something later if the conditions under which privacy can be applied ever change in the future and no one remembers to update this code.

Reply:
Agreed, now using only `supportsPrivacyProtectionPurchase`.